### PR TITLE
Adding configuration class

### PR DESCRIPTION
**Pull request by leobago**
Starting a module that will later receive a configuration file including all the configuration parameters.

### CODE REVIEW

File: study.py [hunk 1]
```diff
@@ -5,31 +5,34 @@
...
+            sim.resetChi(chi)
+            blockAvailable = 0
+            for run in range(config.runStart, config.runStop, config.runStep):
+                sim.logger.info("FR: %d %%, Chi: %d %%, Run: %d ..." % (fr, chi, run), extra=sim.format)
+                sim.initValidators()
+                sim.initNetwork()
+                result = sim.run()
+                blockAvailable += result.blockAvailable
+                results.append(result)
+                simCnt += 1
```
Comment (cskiraly) [inline study.py:29]: I think I would prefer calling a Configuration what is processed by a single run, and use a list of configurations or a generator function to drive the study. That means chi and run would be part of the configuration, while chiStart/chiStop/chiStep would be one possible implementation of parameter space exploration. If you think it sounds good we can change it like that, but it is also OK like it is now, just a bit less generic.

File: study.py [hunk 2]
```diff
@@ -5,31 +5,34 @@
...
-        result = 0
-        for i in range(maxTries):
-            sim.initValidators()
-            sim.initNetwork()
-            result += sim.run()
-            simCnt += 1
+        for chi in range(config.chiStart, config.chiStop+1, config.chiStep):
+            sim.resetChi(chi)
+            blockAvailable = 0
+            for run in range(config.runStart, config.runStop, config.runStep):
```
Comment (cskiraly) [inline study.py:22]: for long simulations it is better (at least in my experience) to make run the external cycle. That means you start to see approx. but complete results earlier, and get more statistical relevance as time passes.

Comment (leobago) [reply study.py]: Done. I also added iterating over the number of validators, network degree and block size.

File: study.py (hunk 1 above)
Comment (leobago) [reply study.py]: I ended up creating a new class called Shape that stores the "shape" of the current simulation, in other words, the exact parameters used in the current simulation. For runs with multiple simulations, I added a function to reset the shape with the new parameters. The configuration is now obtained from a configuration file that is passed by a parameter.

Comment (cskiraly) [reply study.py]: Thanks, sounds great!

File: study.py [hunk 3]
```diff
@@ -1,35 +1,45 @@
...
-            simCnt += 1
-        frRange.append(fr)
-        resultRange.append((maxTries-result)*100/maxTries)
+
+    for run in range(config.numberRuns):
+        for fr in range(config.failureRateStart, config.failureRateStop+1, config.failureRateStep):
+            for chi in range(config.chiStart, config.chiStop+1, config.chiStep):
+                for blockSize in range(config.blockSizeStart, config.blockSizeStop+1, config.blockSizeStep):
+                    for nv in range(config.nvStart, config.nvStop+1, config.nvStep):
+                        for netDegree in range(config.netDegreeStart, config.netDegreeStop, config.netDegreeStep):
```
Comment (cskiraly) [inline study.py:26]: netDegreeStop is missing a +1

**Review by cskiraly — APPROVED** [review]
I see one "+1" missing, see detailed comment. 
Otherwise I think it is good to be merged.

I see some space for streamlining in next steps, e.g. using tuples for start/stop/step.

**Pull request merged**